src/dct/ai/Commander.lua

```diff
@@ -121,16 +121,17 @@ local invalidXpdrTbl = {
 --]]
 function Commander:genMissionCodes(msntype)
 	local id
+	local missionId
 	local digit1 = enum.squawkMissionType[msntype]
 	while true do
-		MISSION_ID = (MISSION_ID + 1) % 64
-		id = string.format("%01o%02o0", digit1, MISSION_ID)
+		missionId = math.random(1, 63)
```

Comment:
This has the potential to loop for a very long time. As the number of sorties increases the density of the table will become greater increasing the probability of a collision. A better algorithm for selecting a free ID in a predictable amount of time should be used. And what happens if we run out of IDs? Currently these IDs are used as the index into the mission table, so they must be unique.

I would recommend randomly selecting the beginning then sequentially looping over all ids until a free one is found or we visit every id. If the id space is exhausted we deny the creation of the mission.

pseudo code:
```
missionid = math.random
testcnt = 1
while true do
    if testcnt > 64 then
        return nil
    end
    id = string.format(...)
    if not in invalidtbl[id] and getMission(id) == nil then
        break
    end
    missionid = (missionid + 1) % 64
    testcnt = testcnt + 1
end
```

This limits the loop to 64 iterations.

Reply:
That is not random, it goes back to the sequential IDs of before where one can effectively traverse all the active missions by trying codes near theirs.

Alternative approach using a shuffled list to determine the ID order: https://github.com/jtoppins/dct/pull/236/commits/00fe7ca75be4ea10807ca68948fe29d88a976145 (only drawback is that it can wrap, but I think it will rarely do so)

Reply:
Given the join feature in the F-10 menu, how does having the mission code random help anyway? If the intent is to have a pseudo random mission ID to prevent randos from joining your mission, doesn't putting all the active mission ids in the F-10 menu defeat this concept?

Reply:
The menu doesn't list on active missions, it lists all possible codes.